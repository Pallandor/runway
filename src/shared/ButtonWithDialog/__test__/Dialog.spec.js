import React from 'react';
import { shallow } from 'enzyme';
import Dialog from '../components/Dialog';

jest.mock('shortid', () => ({ generate: () => 'mockId' }));

describe('Dialog', () => {
  let component;

  const requiredProps = {
    content: 'mock content',
    renderHeader: () => <div>mockHeader</div>,
    renderFooter: () => <div>mockHeader</div>,
    dialogAriaLabel: 'mock aria label',
    dialogStyles: {},
    transitionStyles: { entering: '', entered: '', exiting: '' }
  };

  it('renders correctly with defaults', () => {
    component = shallow(<Dialog {...requiredProps} />);

    expect(component).toMatchSnapshot();
  });

  it('renders correctly with props provided', () => {
    const additionalProps = {
      transitionState: 'mock transition state',
      contentPadding: 'mock content padding',
      lockBgScroll: true
    };
    component = shallow(<Dialog {...requiredProps} {...additionalProps} />);

    expect(component).toMatchSnapshot();
  });
});
