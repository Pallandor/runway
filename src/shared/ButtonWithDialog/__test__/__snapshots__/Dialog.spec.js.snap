// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog renders correctly with defaults 1`] = `
.emotion-0 {
  overflow-y: auto;
  -webkit-overflow-scrolling: touch;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div
  aria-label="mock aria label"
  className="emotion-1"
  role="dialog"
>
  <div>
    mockHeader
  </div>
  <div
    className="emotion-0"
    id="mockId"
  >
    mock content
  </div>
  <div>
    mockHeader
  </div>
</div>
`;

exports[`Dialog renders correctly with props provided 1`] = `
.emotion-0 {
  padding: mock content padding;
  overflow-y: auto;
  -webkit-overflow-scrolling: touch;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div
  aria-label="mock aria label"
  className="emotion-1"
  role="dialog"
>
  <div>
    mockHeader
  </div>
  <div
    className="emotion-0"
    id="mockId"
  >
    mock content
  </div>
  <div>
    mockHeader
  </div>
</div>
`;
